src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdNewCaseCreator.java

```diff
@@ -0,0 +1,212 @@
+
+import feign.FeignException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.InvalidCaseDataException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.TransformationClient;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.request.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.response.CaseCreationDetails;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.transformation.model.response.SuccessfulTransformationResponse;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.CallbackException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.callback.ProcessResult;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.payments.PaymentsPublishingException;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+@Service
+public class CcdNewCaseCreator {
+    private static final Logger log = LoggerFactory.getLogger(CcdNewCaseCreator.class);
+
+    private static final String EXCEPTION_RECORD_REFERENCE = "bulkScanCaseReference";
+
+    private final TransformationClient transformationClient;
+    private final AuthTokenGenerator s2sTokenGenerator;
+    private final PaymentsProcessor paymentsProcessor;
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final ExceptionRecordFinalizer exceptionRecordFinalizer;
+
+    public CcdNewCaseCreator(
+        TransformationClient transformationClient,
+        AuthTokenGenerator s2sTokenGenerator,
+        PaymentsProcessor paymentsProcessor,
+        CoreCaseDataApi coreCaseDataApi,
+        ExceptionRecordFinalizer exceptionRecordFinalizer
+    ) {
+        this.transformationClient = transformationClient;
+        this.s2sTokenGenerator = s2sTokenGenerator;
+        this.paymentsProcessor = paymentsProcessor;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.exceptionRecordFinalizer = exceptionRecordFinalizer;
+    }
+
+    @SuppressWarnings({"squid:S2139", "unchecked"}) // squid for exception handle + logging
+    public ProcessResult createNewCase(
+        ExceptionRecord exceptionRecord,
+        ServiceConfigItem configItem,
+        boolean ignoreWarnings,
+        String idamToken,
+        String userId,
+        CaseDetails exceptionRecordData
+    ) {
```

Comment:
This method is way too long. I realise it was like this in its original version and we should make improvements one step at a time, so this is fine for this PR. Perhaps something for the future, though.